Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt

```diff
@@ -177,7 +187,31 @@ class SubmissionResultPositiveOtherWarningFragment : Fragment(),
     // InternalExposureNotificationPermissionHelper - callbacks
     override fun onKeySharePermissionGranted(keys: List<TemporaryExposureKey>) {
         super.onKeySharePermissionGranted(keys)
-        submissionViewModel.submitDiagnosisKeys(keys)
+        if (keys.isNotEmpty()) { submissionViewModel.submitDiagnosisKeys(keys)
+        } else { showSubmitErrorDialog()
+        }
+    }
+
+    /**
+     * Error Dialog
+     *
+     * If Diagnosis Key submission is failed, this error dialog is shown
+     * Cancellable is set to false
+     *
+     * @see SubmissionViewModel.submissionState
+     * @see AlertDialog.setCancelable
+     */
+    private fun showSubmitErrorDialog() {
+        val submitErrorDialogDialogInstance = DialogHelper.DialogInstance(
+                requireActivity(),
+                R.string.submission_error_dialog_diagnosis_key_title,
+                R.string.submission_error_dialog_diagnosis_key_paired_body,
+                R.string.submission_error_dialog_diagnosis_key_paired_button_positive,
```

Comment:
same here, can we make it obvious that this is for the no diagnosis keys case?

Reply:
made the necessary changes